frontend/src/components/Downloader.jsx

```diff
@@ -0,0 +1,84 @@
+import React, { Component } from "react";
+import axios from "axios";
+import { Modal, message, Progress } from "antd";
+import JSZip from "jszip";
+import { saveAs } from "file-saver";
+
+class Downloader extends Component {
+  constructor() {
+    super();
+    this.state = {
+      progress: 0
+    };
+  }
+
+  componentDidMount() {
+    this.downloadImages();
+  }
+
+  downloadImages = () => {
+    const { links } = this.props;
+    let count = 0;
+    let zip = JSZip();
+
+    console.log(links);
+
+    links.forEach(l => {
+      axios
+        .get(l, {
+          responseType: "blob"
+        })
+        .then(res => {
+          let imageType = l.split(".")[1];
+          zip.file(`image${count}.${imageType}`, res.data, {
+            binary: true
+          });
+
+          ++count;
+          this.setState({ progress: (count / links.length) * 100 });
+
+          if (count === links.length) {
```

Comment:
Shouldn't this part be moved to outside the forEach? That way we won't have to check `count === links.length` everytime.